### PR TITLE
Fuzz tests for OCSP parser

### PR DESCRIPTION
**Pull request by samuel40791765**
### Issues:
Resolves CryptoAlg-753

### Description of changes: 
- Added fuzzing guide on how to add fuzz tests.
- Added fuzz test and fuzzing corpus for new OCSP parsing functionality. Files under `fuzz/ocsp_corpus` were generated by an initial 30 minute run of the fuzzer on our OCSP unit tests.

### Call-outs:
N/A

### Testing:
Added another OCSP fuzzing dimensions to our fuzz tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: fuzz/ocsp.cc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    // Reserialize the structure.
+    uint8_t *der = NULL;
+    i2d_OCSP_RESPONSE(ocsp_response, &der);
+    OPENSSL_free(der);
+  }
+  OCSP_RESPONSE_free(ocsp_response);
+  ERR_clear_error();
+  return 0;
+}
```
Comment (bryce-shang) [inline fuzz/ocsp.cc:26]: Some questions:

1. Is the input (`const uint8_t *buf, size_t len`) from `fuzz/ocsp_corpus` dir?

2. Is this file used to tell fuzz how to test OCSP related functions (e.g. `d2i_OCSP_RESPONSE`, `OCSP_response_get1_basic`)?

File: FUZZING.md [hunk 2]
```diff
@@ -80,3 +80,47 @@ If both sets of tests pass, refresh the fuzzer corpora with `refresh_ssl_corpora
...
+
+## Adding New Fuzz Test Targets
+When adding new functionality, adding new fuzz tests are important to provide additional testing and verification that we are correct.
+
+### Steps
+1. `NEW_FUNCTION='name_of_new_fuzzing_target'`
+2. `touch fuzz/${NEW_FUNCTION}.cc` \
+    Write fuzzing code for libfuzzer to parse in `fuzz/${NEW_FUNCTION}.cc`. 
+3. `mkdir fuzz/${NEW_FUNCTION}_corpus_raw` \
+   Import unit test files into the `fuzz/${NEW_FUNCTION}_corpus_raw` directory. These will be used to create our new fuzzing corpus.
```
Comment (bryce-shang) [inline FUZZING.md:92]: > Import unit test files...

Where are the unit test files originally from?

File: FUZZING.md [hunk 3]
```diff
@@ -80,3 +80,47 @@ If both sets of tests pass, refresh the fuzzer corpora with `refresh_ssl_corpora
...
 ./refresh_ssl_corpora.sh /path/to/fuzzer/mode/build /path/to/non/fuzzer/mode/build
 ```
+
+## Adding New Fuzz Test Targets
+When adding new functionality, adding new fuzz tests are important to provide additional testing and verification that we are correct.
+
+### Steps
+1. `NEW_FUNCTION='name_of_new_fuzzing_target'`
+2. `touch fuzz/${NEW_FUNCTION}.cc` \
+    Write fuzzing code for libfuzzer to parse in `fuzz/${NEW_FUNCTION}.cc`.
```
Comment (bryce-shang) [inline FUZZING.md:90]: Is the `parse(serialize and deserialize)` the only purpose of step 2? I might have added duplicate questions in `fuzz/ocsp.cc`.

File: fuzz/ocsp_corpus/01cd28b0c1554e1a7207586d00869269eb08c66c [hunk 4]
```diff
@@ -0,0 +1 @@
+����������
```
Comment (bryce-shang) [inline fuzz/ocsp_corpus/01cd28b0c1554e1a7207586d00869269eb08c66c:1]: Does this file just include binaries like `fuzz/ocsp_corpus/01f48d134e3943f77accb2ec7595d4bd6791139a`, which is marked by GitHub with `Binary file not shown.`?

File: fuzz/ocsp.cc (hunk 1 above)
Comment (samuel40791765) [reply fuzz/ocsp.cc:26]: 1. Yes, this input reads from directory we give to the fuzzing command. In this case, it's `fuzz/ocsp_corpus`.
2. Yes, this file is how we tell Libfuzzer to parse our inputs. I'll clarify that in the runbook guide.

File: FUZZING.md (hunk 2 above)
Comment (samuel40791765) [reply FUZZING.md]: I got the original unit tests from our current OCSP test files. I also converted the `ors` files to a hex format, so the `d2i_OCSP_Response` could correctly parse it. 
The current `fuzz/ocsp_corpus` was generated from a 30 minute fuzzing run on those unit tests.

File: FUZZING.md (hunk 3 above)
Comment (samuel40791765) [reply FUZZING.md]: Yes, I'll clarify that in the fuzzing runbook. I initially had the same question when first creating the fuzzing test file.

**Review by bryce-shang — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

**Pull request merged**